tests/test-mongoc-uri.c

```diff
@@ -96,12 +98,15 @@ test_mongoc_uri_new (void)
    ASSERT_CMPSTR(bson_iter_utf8(&iter, NULL), "1");
    mongoc_uri_destroy(uri);
 
-   uri = mongoc_uri_new("mongodb://localhost:27017/?readPreferenceTags=dc:ny&readPreferenceTags=");
+   uri = mongoc_uri_new("mongodb://localhost:27017/?readPreference=secondaryPreferred&readPreferenceTags=dc:ny&readPreferenceTags=");
```

Comment:
Since we now validate the read preference, I needed to specify something other than `primary` (the default) here to ensure URI construction actually succeeds.